src/navigation/reroute.js

```diff
@@ -109,74 +116,79 @@ export function reroute(pendingPromises = [], eventArguments) {
         )
       );
 
-      if (navigationIsCanceled) {
-        window.dispatchEvent(
-          new CustomEvent(
-            "single-spa:before-mount-routing-event",
-            getCustomEventDetail(true)
-          )
-        );
-        finishUpAndReturn();
-        navigateToUrl(oldUrl);
-        return;
-      }
-
-      const unloadPromises = appsToUnload.map(toUnloadPromise);
-
-      const unmountUnloadPromises = appsToUnmount
-        .map(toUnmountPromise)
-        .map((unmountPromise) => unmountPromise.then(toUnloadPromise));
-
-      const allUnmountPromises = unmountUnloadPromises.concat(unloadPromises);
+      return Promise.all(cancelPromises).then((results) => {
+        if (results.indexOf(true) !== -1) {
```

Comment:
A truthiness check here would probably be more intuitive for most users:

```suggestion
        if (cancelValues.some(v => v)) {
```